config/m4/compiler.m4

```diff
@@ -84,15 +85,66 @@ AC_DEFUN([COMPILER_OPTION],
     AS_IF([test "x$with_$1" != "xno"],
           [SAVE_CFLAGS="$CFLAGS"
            CFLAGS="$BASE_CFLAGS $CFLAGS $3"
-           AC_MSG_CHECKING([$2])
+           AC_MSG_CHECKING([$3])
            CHECK_CROSS_COMP([AC_LANG_SOURCE([$5])],
                             [AC_MSG_RESULT([yes])
-                             OPT_CFLAGS="$OPT_CFLAGS|$1"],
+			     AS_IF([test "x$1" != "xmcpu" -a "x$1" != "xmarch"],
+                             [OPT_CFLAGS="$OPT_CFLAGS|$1"],[])],
```

Comment:
fix UCS CPU check to ignore some flags, or even better - extend the runtime check to test them as well

Reply:
I was thinking about this but overall it is more elegant solution that fixing x86 specific assumptions in the code. I will have separate PR (eventually..) that will verify the features for arm.

Reply:
so maybe add TODO here?

Reply:
Done. Are we good to go ?